### PR TITLE
BUG make 'sequential' the default remote backend

### LINKED ISSUE #206: Civis joblib backend can ignore n_jobs restriction
In v1.7, the Civis joblib backend gained the ability to set itself as the backend inside child jobs. This means that child jobs which use joblib could also use the Civis joblib backend to paralellize themselves in different container scripts. These child jobs would be unaware of any jobs limit which the user set on the original joblib call, and wouldn't have a way to coordinate between themselves in any case. This could lead to users consuming drastically more resources than they expected.

A further two points: 
- The addition of this feature in v1.7 defaulted to setting the remote backend to "civis", which altered the previous behavior of the Civis joblib backend (it didn't launch jobs from inside jobs before). 
- This behavior differs from the way that joblib treats parallel computation with multiprocessing. Attempting to multiprocess from inside a multiprocessed job results in ~~an error~~ joblib defaulting to running serially.

I think that in principle, the Civis joblib backend is doing what it should -- resources in the cloud aren't limited in the same way that resources on a local machine are. But the inability to control resource use isn't good, and the hard-to-notice change in default behavior also isn't good. We should switch to "threading" as the default `remote_backend` so that users need to request the new, more aggressive behavior (and manage the total number of jobs through `n_jobs` parameters inside whatever they're calling). It would be much more difficult to try to make the `n_jobs` parameter on the Civis joblib backend actually restrict the total number of jobs when using `remote_backend = "civis"`, so for now I think we should settle for documenting.

### PR DESCRIPTION
**Pull request by beckermr**
Before we used 'civis' as the default. This means that containers are viral. Using the threading default is probably less confusing/surprising. IDK if we had this discussion before, but whatever the case, I am now convinced this is a better choice.

Closes #206 

@jmemich for viz

### CODE REVIEW

**Comment by beckermr** [comment]
@elsander If we merge this PR and make this the default, then you would need to set the remote backend to 'civis' for CivisML.

**Comment by stephen-hoover** [comment]
This is backwards incompatible, so it would have to wait on a v2 release. I'm also not convinced we should do this. It's worth an in-person discussion.

**Comment by stephen-hoover** [comment]
Please also submit PRs from forks, rather than creating branches on the main repo.

**Comment by beckermr** [comment]
This is actually *reverting* a backwards incompatible change from adding this feature in v1.7.0.

**Comment by beckermr** [comment]
Sorry about not using a fork!

**Comment by beckermr** [comment]
To be more clear, the previous behavior of backend was to not launch containers for nest Parallel calls. When we added this feature and set the default, we then changed this behavior to launch them in nested parallel calls by default. This PR changes the default behavior back to what would happen in 1.6.2 or earlier. FWIW, the new default surprised @jmemich and I personally forgot about it in several places.

**Comment by beckermr** [comment]
@stephen-hoover I think this is ready to go then for CR.

**Comment by beckermr** [comment]
Sure will do!

**Comment by beckermr** [comment]
OK @stephen-hoover This should be good to go. Back to you!

File: civis/parallel.py [hunk 1]
```diff
@@ -280,8 +283,11 @@ def make_backend_factory(docker_image_name="civisanalytics/datascience-python",
     remote_backend : str or object, optional
         The name of a joblib backend or a joblib backend itself. This parameter
         is the joblib backend to use when executing code within joblib in the
-        container. The default of 'civis' uses an exact copy of the Civis
-        joblib backend that launched the container.
+        container. The default of 'sequential' uses the joblib sequential
+        backend in the container. The value 'civis' uses an exact copy of the
+        Civis joblib backend that launched the container.Note that with the
```
[the remark this answers is not in the document]
Comment (beckermr) [reply civis/parallel.py]: (facepalm)

**Comment by beckermr** [comment]
OK @stephen-hoover. Let's try this one more time...

**Pull request merged**